src/github.com/getlantern/balancer/dialer.go

```diff
@@ -52,12 +52,15 @@ var (
 
 type dialer struct {
 	*Dialer
-	active int32
-	errCh  chan time.Time
+	active  int32
+	closeCh chan interface{}
+	errCh   chan time.Time
 }
 
 func (d *dialer) start() {
 	d.active = 1
+	// to avoid blocking sender, make it buffered
+	d.closeCh = make(chan interface{}, 1)
```

Comment:
Use a separate channel to tell `dialer` to close itself. Avoid panic of sending to closed 'errCh'